gym_electric_motor/visualization/motor_dashboard_plots.py

```diff
@@ -0,0 +1,386 @@
+            self.update()
+            self._axis.axvline(self._t, color='red', linewidth=1)
+
+    def update(self):
+        self._reward_line.set_data(self._t_data, self._reward_data)
+        if self.mode == 'continuous':
+            x_lim = self._axis.get_xlim()
+            upper_lim = max(self._t, x_lim[1])
+            lower_lim = upper_lim - self.x_width
+            self._axis.set_xlim(lower_lim, upper_lim)
+
+
+class ActionPlot(MotorDashboardPlot):
+    """ class to plot the instantaneous actions applied on-to the environment
+
+    """
+    x_width = 3
+    mode = 'continuous'
+    action_line_cfg = {
+        'color': 'magenta',
+        'linestyle': '',
+        'linewidth': 0.75,
+        'marker': '.',
+        'markersize': .5
+    }
+
+    def __init__(self, action):
+        super().__init__()
+        # the action space of the environment. can be Box() or Discrete()
+        self._action_space = None
+        self._action = action
+        # the index for the actions.
+        self._action_idx = None
+        self._action_line = None
+        # Data containers
+        self._action_data = None
+        self._t_data = None
+        self._tau = None
+        self._x_points = None
+        self._t = 0
+        # the range of the action values
+        self._action_range_min = None
+        self._action_range_max = None
+        # the type of action space: Discrete or Continuous
+        self._action_type = None
+
+    def initialize(self, axis):
+        """
+        Args:
+            axis (object): the subplot axis for plotting the action variable
+        """
+        super().initialize(axis)
+        axis.grid()
+
+        self._t_data = np.linspace(0, self.x_width, self._x_points, endpoint=False)
+        self._action_data = np.zeros_like(self._t_data, dtype=float)
+        self._action_line, = self._axis.plot(self._t_data, self._action_data, **self.action_line_cfg)
+        # set the layout of the subplot
+        act_min = self._action_range_min
+        act_max = self._action_range_max
+        spacing = (act_min - act_max) * 0.1
+        self._axis.set_ylim(act_min - spacing, act_max + spacing)
+        self._axis.set_xlim(0, self.x_width)
+        #self._axis.set_xlabel('t/s')
+        self._axis.set_ylabel(self._action)
+        base_action_line = lin.Line2D([], [], color=self.action_line_cfg['color'])
+        self._axis.legend((base_action_line,), (self._action,), loc='upper left')
+
+    def set_modules(self, ps, rg, rf):
+        # fetch the action space from the physical system
+        self._action_space = ps.action_space
+        #extract the action index from the action name
+        self._action_idx = int(self._action.split('_')[1])
+        # check for the type of action space: Discrete or Continuous
+        if self._action_space.dtype == 'float32':  # for continuous action space
+            self._action_type = 'Continuous'
```

Comment:
self._mode goes with lower case "continuous" while self._action_type goes with CapitalCase "Continuous". Lets have everything in lower case